drivers/pwm/pwm_mspm0.c

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2024 Bang & Olufsen A/S, Denmark
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_mspm0_pwm
+
+#include <ti/driverlib/dl_timera.h>
+#include <ti/driverlib/dl_timerg.h>
+#include <zephyr/drivers/pwm.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/logging/log.h>
+
+#define REG_TIMA0            (GPTIMER_Regs *)0x40860000
+#define REG_TIMA1            (GPTIMER_Regs *)0x40862000
+#define IS_TIMA(tim)         (tim == REG_TIMA0 || tim == REG_TIMA1)
+// from the datasheet, only TIMA0 has 4 capture compare channels
+#define HAS_FOUR_CC(tim)     (tim == REG_TIMA0)
+#define MAX_ALLOWED_CHANNELS_TIMX         2
+#define MAX_ALLOWED_CHANNELS_WITH_FOUR_CC 4
```

Comment:
```suggestion
#define NUM_CHANNELS(tim) (HAS_FOUR_CC(tim) ? 4 : 2) 
```
You could make these a function like macro. I think it'll make the callsites cleaner while conveying the same info wrt. defaulting to 2 channels